docs/ubuntu_guide.md

```diff
@@ -0,0 +1,170 @@
+# Trin on Ubuntu: Setup guide
+
+These steps are for setting up a Trin node as a service on Ubuntu.
+
+## Installation
+```
+$ sudo apt install libssl-dev librocksdb-dev libclang-dev pkg-config build-essentials
+```
+Install Trin:
+> Tip: If you intend to submit code changes to trin, first fork the repo and
+then clone that url.
+```
+$ cd ~
+$ git clone https://github.com/ethereum/trin.git
+$ cd trin
+$ cargo build --workspace --release
+```
+Now the executable is located in `trin/target/release` and can be called by systemd.
+Move that binary to the standard location for binaries:
+```
+$ sudo cp -a ~/trin/target/release/trin /usr/local/bin/trin
+```
+> Tip: If you make changes to these steps, keep a record for future reference.
+
+Make a new user for the Trin service:
+```
+$ sudo useradd --no-create-home --shell /bin/false trin
+```
+Make a directory for Trin data and give the Trin user permission to access it:
+```
+$ sudo mkdir -p /var/lib/trin
+$ sudo chown -R trin:trin /var/lib/trin
+```
+Check that the binary works:
+```
+$ /usr/local/bin/trin --version
+```
+Example response:
+```
+> Launching trin
+> trin 0.0.1
+```
+## Configuration
+Before setting up the service, look at the flags can be set when starting Trin.
+```
+$ /usr/local/bin/trin --help
+```
+### Optional flag for database size
+`--kb 200000`. Trin lets you control how much storage the node takes up (e.g., 200MB). The default is
+100_000 kilobytes and can be changed.
+
+### Optional flag for no connection to external server
+
+`--no-stun`. A third party server connection is configured by default to assist in testing.
+This is a Session Traversal Utilities for NAT (STUN) server and may be disabled
+a flag. The docs state: "Do not use STUN to determine an external IP. Leaves
+ENR entry for IP blank. Some users report better connections over VPN."
+
+### Optional flag for conflicting nodes
+
+`--discovery-port <port>`. If an Ethereum execution node is already running, it may be using
+the default port 8545. To pick a new port, select a number in the range 1024–49151 and
```

Comment:
I think the default port 8545 is used for JSON-RPC not for discovery, which `--discovery-port <port>` points to? However, we still can get some conflicts on the TCP 8545 because the default portal json-rpc port is same as eth json-rpc, so changing it with  `--web3-http-address` is reasonable.

Reply:
Oh you are right, I have conflated discovery and rpc here